datafusion/physical-expr/src/expressions/nullif.rs

```diff
@@ -17,7 +17,7 @@
 
 use arrow::array::Array;
 use arrow::compute::eq_dyn;
-use arrow::compute::nullif::nullif;
+use arrow_select::nullif::nullif;
```

Comment:
Upstream fix - https://github.com/apache/arrow-rs/pull/3451

Longer term I hope to move datafusion away from the `arrow` dependency, and so this isn't really the end of the world